src/main/kotlin/fi/metatavu/metaform/server/controllers/TemplateController.kt

```diff
@@ -0,0 +1,135 @@
+@ApplicationScoped
+class TemplateController {
+
+    @Inject
+    lateinit var templateDAO: TemplateDAO
+
+    @Inject
+    lateinit var logger: Logger
+
+    /**
+     * Creates new template
+     *
+     * @param templateData TemplateData
+     * @param visibility TemplateVisibility
+     * @param creatorId UUID
+     * @return created template
+     */
+    fun createTemplate(
+            templateData: TemplateData,
+            visibility: TemplateVisibility,
+            creatorId: UUID
+    ): Template {
+
+        val serializedTemplateData = try {
+            serializeTemplateData(templateData)
+        } catch (e: Exception) {
+           throw Exception(e.message)
+        }
+
+        return templateDAO.create(
+                id = UUID.randomUUID(),
+                data = serializedTemplateData,
+                visibility = visibility,
+                creatorId = creatorId,
+                lastModifierId = creatorId
+        )
+    }
+
+    /**
+     * Finds a template by id
+     *
+     * @param id id
+     * @return found template or null if not found
+     */
+    fun findTemplateById(id: UUID): Template? {
+        return templateDAO.findById(id)
+    }
+
+    /**
+     * Updates a template
+     *
+     * @param template template
+     * @param data data
+     * @return updated template
+     */
+    fun updateTemplate(
+        template: Template,
+        data: String,
+        templateVisibility: TemplateVisibility,
+        lastModifier: UUID
+    ): Template {
+        return templateDAO.updateData(
+            template = template,
+            data = data,
+            templateVisibility = templateVisibility,
+            lastModifier = lastModifier
+        )
+    }
+
+    /**
+     * Deletes an template
+     *
+     * @param template template to be deleted
+     */
+    fun deleteTemplate(template: Template) {
+        templateDAO.delete(template)
+    }
+
+    /**
+     * Lists templates by visibility
+     * @param visibility TemplateVisibility
+     */
+    fun listTemplates(visibility: TemplateVisibility? = null): List<Template> {
+        visibility ?: return templateDAO.listAll()
+        return templateDAO.listByVisibility(visibility)
+    }
+
+    /**
+     * Serializes TemplateData as string
+     *
+     * @param templateData TemplateData
+     * @return data as string
+     */
+
+    @Throws(Exception::class)
+    fun serializeTemplateData(templateData: TemplateData): String {
+    /*    try {
+            return jacksonObjectMapper().writeValueAsString(templateData)
+        } catch (e: Exception) {
+            throw Exception("Failed to serialize template data", e)
+        }
+
+     */
+        val objectMapper = ObjectMapper()
+        objectMapper.registerModule(JavaTimeModule())
+
+        try {
+            return objectMapper.writeValueAsString(templateData)
+        } catch (e: JsonProcessingException) {
+            throw MalformedMetaformJsonException("Failed to serialize draft data", e)
```

Comment:
MalformedMetaformJsonException is used when system detects malformed "metaform" JSON, create new dedicated exception (e.g. MalformedTemplateJsonException) instead